spirv_hlsl.cpp

```diff
@@ -1102,26 +1102,52 @@ void CompilerHLSL::emit_texture_op(const Instruction &i)
 		texop += "texelFetch";
 	else
 	{
-		texop += "tex2D";
-
-		if (gather)
-			texop += "Gather";
-		if (coffsets)
-			texop += "Offsets";
-		if (proj)
-			texop += "Proj";
-		if (grad_x || grad_y)
-			texop += "Grad";
-		if (lod)
-			texop += "Lod";
+		if (options.shader_model >= 40)
+		{
+			texop += to_expression(img);
+
+			if (gather)
+				texop += ".Gather";
+			else if (offset || coffsets || bias)
+				texop += ".SampleBias";
+			else if (proj)
+				SPIRV_CROSS_THROW("textureProj is not supported in HLSL shader model 4/5.");
```

Comment:
I think you'd have to manually divide xyz / w. That's basically how textureProj is defined anyways.